ipsec_offload_plugin/ipsec_offload/ipsec_offload.c

```diff
@@ -945,20 +938,26 @@ METHOD(kernel_ipsec_t, del_policy, status_t,
 							       data->sa->mode == MODE_TUNNEL);
 			if(err == IPSEC_FAILURE)
 				DBG2(DBG_KNL, "Inline_crypto_ipsec ipsec_tx_sa_classification_table:"
-				     "add entry failed");
+				     "del entry failed");
 		} else {
 			reqid_bitclear(data->sa->reqid);
 		}
 
 		if (data->sa->mode == MODE_TUNNEL) {
+			
```

Comment:
The dcap is removed. The encap remains!!! Would like to see the P4. Also as we discussed the MAC should be handled internally. So the encap should be removed.

Reply:
MAC is getting handled through krnlmon and OVS. Encap table is used for setting IPSEC_ENCAP mod action